streams/src/test/java/org/apache/kafka/streams/processor/internals/StreamsPartitionAssignorTest.java

```diff
@@ -1286,7 +1286,10 @@ public void shouldReturnUnchangedAssignmentForOldInstancesAndEmptyAssignmentForF
             )));
         assertThat(assignment.get("consumer1").partitions(), equalTo(Utils.mkList(t1p0, t1p1)));
 
-        assertThat(AssignmentInfo.decode(assignment.get("future-consumer").userData()), equalTo(new AssignmentInfo()));
```

Comment:
Why is this change required? It seems we forgot to update `AssignmentInfo#equals()` and `AssignmentInfo#hashCode()`...